base_layer/core/src/transactions/transaction_protocol/transaction_initializer.rs

```diff
@@ -526,15 +525,13 @@ impl SenderTransactionInitializer {
 
             // If rewind data is present we produce a rewindable output, else a standard output
             let change_output = if let Some(rewind_data) = self.rewind_data.as_ref() {
-                // TODO: Should proof be verified?
```

Comment:
Personally, I don't think it needs to, the code that generates it is a few lines up and any discrepancy between the way the proof is constructed here and verified in other parts of the code base will be picked up in the various tests.